types/trace-events.d.ts

```diff
@@ -0,0 +1,590 @@
+    FunctionCall.E |
+    InvalidateLayout.I |
+    Layout.B |
+    Layout.E |
+    LoadEventEnd.R |
+    NavigationStart.R |
+    ParseAuthorStyleSheet.X |
+    Process_labels.M |
+    RequestStart.R |
+    ResourceFinish.I |
+    ResourceReceiveResponse.I |
+    ResourceSendRequest.I |
+    RunTask.X |
+    ScheduleStyleRecalculation.I |
+    Screenshot.O |
+    Thread_name.M |
+    ThreadControllerImpl.DoWork.X |
+    ThreadControllerImpl.RunTask.X |
+    TimerFire.X |
+    TimerInstall.I |
+    TracingStartedInBrowser.I |
+    V8.Compile.B |
+    V8.Compile.E |
+    V8.Compile.X |
+    XHRReadyStateChange.X;
+
+  namespace DomContentLoadedEventEnd {
+    interface R extends Base {
+      args: {
+        frame: string;
+      };
+      name: 'domContentLoadedEventEnd';
+      ph: 'R';
+      tts: number;
+    }
+  }
+
+  namespace EvaluateScript {
+    interface X extends Base {
+      args: {
+        data?: {
+          columnNumber: number;
+          frame: string;
+          lineNumber: number;
+          stackTrace?: {
+            columnNumber: number;
+            functionName: string;
+            lineNumber: number;
+            scriptId: string;
+            url: string;
+          }[];
+          url: string;
+        };
+      };
+      dur: number;
+      name: 'EvaluateScript';
+      ph: 'X';
+      tdur: number;
+      tts: number;
+    }
+  }
+
+  namespace FirstContentfulPaint {
+    interface R extends Base {
+      args: {
+        data: {
+          navigationId: string;
+        };
+        frame: string;
+      };
+      name: 'firstContentfulPaint';
+      ph: 'R';
+      tts: number;
+    }
+  }
+
+  namespace FirstMeaningfulPaint {
+    interface R extends Base {
+      args: {
+        afterUserInput?: number;
+        data?: {
+          navigationId: string;
+        };
+        frame: string;
+      };
+      name: 'firstMeaningfulPaint';
+      ph: 'R';
+      tts: number;
+    }
+  }
+
+  namespace FirstMeaningfulPaintCandidate {
+    interface R extends Base {
+      args: {
+        data: {
+          navigationId: string;
+        };
+        frame: string;
+      };
+      name: 'firstMeaningfulPaintCandidate';
+      ph: 'R';
+      tts: number;
+    }
+  }
+
+  namespace FirstPaint {
+    interface R extends Base {
+      args: {
+        data: {
+          navigationId: string;
+        };
+        frame: string;
+      };
+      name: 'firstPaint';
+      ph: 'R';
+      tts: number;
+    }
+  }
+
+  namespace FunctionCall {
+    interface B extends Base {
+      args: {
+        data: {
+          columnNumber: number;
+          frame: string;
+          functionName: string;
+          lineNumber: number;
+          scriptId: string;
+          url: string;
+        };
+      };
+      name: 'FunctionCall';
+      ph: 'B';
+      tts: number;
+    }
+  
+    interface E extends Base {
+      args: {
+      
+      };
+      name: 'FunctionCall';
+      ph: 'E';
+      tts: number;
+    }
+  }
```

Comment:
what about exporting both types and namespaces collecting each of these. So it would be something like
```ts
namespace FunctionCall {
  interface B extends Base {
    // ...
  }
  interface E extends Base {
    // ...
  }
}
type FunctionCall = FunctionCall.B | FunctionCall.E;
```
(I think the nesting is right there, but may need tweaking :)

that way if you only care about filtering on `name` you have a convenient shorthand for all the events with that name, and for names like `FirstPaint` which *only* has a `ph: 'R'` event, you don't need the final `.R` in `LH.TraceEvent.FirstPaint.R` to refer to the type.

Reply:
(`type TraceEvent` would also then only need to be the union of the unions, rather than list all B/E/X/Whatever individual types)

Reply:
done